### PR TITLE
Incorporate performance periods into award queue

### PR DESCRIPTION
**Pull request by jim**
## Description

This patch updates the TSP award queue to restrict queries for `TSPPerformance` models to those in the same performance period as the shipment.

## Reviewer Notes

* I added an `award_date` column to the `shipments` table to store the date we use to identify which performance period to operate within. I tried to use `created_at`, but Pop [overwrites this field with the current time without checking if it already has a value](https://github.com/gobuffalo/pop/blob/master/model.go#L140-L145).

  The name of this column is potentially confusing and could be misinterpreted as _when the shipment was awarded_, but I couldn't come up with anything better, and the `ShipmentAward` is where that value would live anyway. 

  The `award_date` column, as its name suggests, is a `date` column in the database. We have not talked about timezones in the project yet, and this works around that concern, but it's worth keeping in mind going forward that a date is ambiguous unless it is coupled with timezone or location info.

* I used `BETWEEN` in a SQL query. We could use `<=` and `>=` instead, but I think this is more readable. Note that:

    >  `BETWEEN` treats the endpoint values as included in the range.

## Verification Steps

* [ ] All tests pass.
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely) have been satisfied.

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/155339297) for this change
* [PostgreSQL docs for `BETWEEN`](https://www.postgresql.org/docs/10/static/functions-comparison.html).

### CODE REVIEW

**Comment by breanneboland** [comment]
I think awarded_date is a little clearer, so far as your area of concern, but I also think it may be some unnecessary lily gilding at this particular point in time. I think, like most of our field names, it'll make good sense in a few days once it's become normal to us. 

Always happy to see when time is taken to optimize this particular thing, though. :)

File: pkg/models/transportation_service_provider_performance_test.go [hunk 1]
```diff
@@ -108,24 +111,25 @@ func (suite *ModelSuite) Test_BVSWithLowMPS() {
...
 	tsp2, _ := testdatagen.MakeTSP(suite.db, "Test TSP 2", "TSP2")
 	tsp3, _ := testdatagen.MakeTSP(suite.db, "Test TSP 3", "TSP2")
+
 	// TSPs should be orderd by award_count first, then BVS.
 	testdatagen.MakeTSPPerformance(suite.db, tsp1, tdl, swag.Int(1), mps+1, 0)
 	testdatagen.MakeTSPPerformance(suite.db, tsp2, tdl, swag.Int(1), mps+3, 0)
 	testdatagen.MakeTSPPerformance(suite.db, tsp3, tdl, swag.Int(1), mps+2, 0)
 
-	tsp, err := NextTSPPerformanceInQualityBand(suite.db, tdl.ID, 1)
+	date, _ := time.Parse("Jan 2, 2006", "May 16, 2019")
```
Comment (akostibas) [inline pkg/models/transportation_service_provider_performance_test.go]: The 2 parameters to `Parse` are super confusing: the fact that they both look like dates, except one of them is a format string is crazy making. Maybe we could have a constant at the top of the file that is our format string, so it's easy to tell which is the date we're actually using.

Comment (akostibas) [reply pkg/models/transportation_service_provider_performance_test.go]: Also, inconsistent date error checking. You catch it on 311, but not here.

Comment (jim) [reply pkg/models/transportation_service_provider_performance_test.go]: Yeah, it's kind of nuts. I added some constants to `testdatagen` that can be used both within that package and in our tests. I think it's a big improvement.

File: pkg/awardqueue/awardqueue_test.go [hunk 2]
```diff
@@ -206,7 +206,6 @@ func Test_getTSPsPerBandNoRemainder(t *testing.T) {
 }
 
 func Test_assignTSPsToBands(t *testing.T) {
-	pop.Debug = true
```
[the remark this answers is not in the document]
Comment (jim) [reply pkg/awardqueue/awardqueue_test.go]: Pretty sure I left that there... 😁

**Comment by jim** [comment]
@breanneboland My concern about using `AwardedDate` is that it implies that the shipment was _already_ awarded, when we're really using the date to determine when it's OK to start trying to award it. The actual awarding could theoretically not happen the same day.

I think there is a missing concept here. The date we use to assign the shipment to a performance period is pretty important. `PerformancePeriodAssignmentDate` is the best I can come up with, but it is a bit of a mouthful.

**Comment by breanneboland** [comment]
@jim: aha, yeah, that changes my perception of it significantly. Scratch that, then!

**Pull request merged**